src/operator/deinput.rs

```diff
@@ -0,0 +1,703 @@
+use crate::{algebra::ZRingValue, CollectionHandle, DBData, DBWeight, InputHandle, UpsertHandle};
+use erased_serde::{deserialize, Deserializer as ErasedDeserializer, Error as EError};
+use serde::Deserialize;
+
+const MAX_REUSABLE_CAPACITY: usize = 100_000;
+
+/// An input handle that deserializes values before pushing them to
+/// a stream.
+///
+/// A trait for a type that wraps around an [`InputHandle`] and
+/// pushes serialized data to an input stream.  The client passes a
+/// deserializer object that provides access to a serialized data
+/// value (e.g., in JSON or CSV format) to this API.  This data
+/// gets deserialized into the strongly typed representation
+/// expected by the input stream and pushed to the circuit using
+/// the `InputHandle` API.
+///
+/// This trait is object-safe, i.e., it can be converted to `dyn DeInputHandle`
+/// and invoked using dynamic dispatch.  This allows choosing the input
+/// format (encapsulated by the deserializer object) at runtime or even
+/// switching between different formats dynamically.  Object safety is
+/// achived by using the `erased_serde` crate that exposes an object-safe
+/// deserializer API ([`erased_serde::Deserializer`]), so that `DeInputHandle`
+/// methods don't need to be parameterized by the deserializer type.
+///
+/// Note: this trait is useful for feeding scalar values to the circuit.
```

Comment:
the naming should reflect this then
Call the other one DeScalarHandleImpl

Reply:
Makes sense.  TBH, I am not happy with the term "scalar".  This handle can be used with values of any type, it's just that it accepts the entire value at once vs the the handle that works record-by-record.